pwndbg/commands/__init__.py

```diff
@@ -230,19 +230,45 @@ def fix(
     )
     assert target, "Reached command expression evaluation with no frame or inferior"
 
+    # Try to evaluate the expression in the local, or, failing that, global
+    # context.
     try:
         return target.evaluate_expression(arg)
     except Exception:
         pass
 
+    ex = None
     try:
+        # This will fail if gdblib is not available. While the next check
+        # alleviates the need for this call, it's not really equivalent, and
+        # we'll need a debugger-agnostic version of regs.fix() if we want to
+        # completely get rid of this call. We can't do that now because there's
+        # no debugger-agnostic architecture functions. Those will come later.
+        #
+        # TODO: Port architecutre functions and `pwndbg.gdblib.regs.fix` to debugger-agnostic API and remove this.
         arg = pwndbg.gdblib.regs.fix(arg)
         return target.evaluate_expression(arg)
     except Exception as e:
+        ex = e
+
+    # If that fails, try to treat the argument as the name of a register, and
+    # see if that yields anything.
+    if frame:
+        regs = frame.regs()
+        arg = arg.strip()
+        if arg.startswith("$"):
+            arg = arg[1:]
+        reg = regs.by_name(arg)
+        if reg:
+            return reg
```

Comment:
I don't get it. Why do we need this?

All this logic here is about `<command> <arguments>` and the handling for each separate argument, right? If so, why do we do `arg.strip()` here at all?

Also, does this mean that llvm doesn't provide us some "pls parse the arguments and giv us debugger value objects" api?

Reply:
This needs to be documented in the code, otherwise its super unclear why we need it in the first place.

Also, if we have `regs.fix` in a later branch, I assume this logic is then not really needed and will be removed?

Reply:
We should probably have the latest change here in this PR or just merge it and go fast with newer PRs. Lets do the 2nd.

Reply:
> This needs to be documented in the code, otherwise its super unclear why we need it in the first place.

Fair enough, I can make that clearer, then.

> Also, if we have `regs.fix` in a later branch, I assume this logic is then not really needed and will be removed?

Yes, it's a temporary solution until we get `regs.fix` working under LLDB.